cmd/compose/run.go

```diff
@@ -148,15 +149,25 @@ func runCommand(p *projectOptions, backend api.Service) *cobra.Command {
 	flags.StringVarP(&opts.workdir, "workdir", "w", "", "Working directory inside the container")
 	flags.StringVar(&opts.entrypoint, "entrypoint", "", "Override the entrypoint of the image")
 	flags.BoolVar(&opts.noDeps, "no-deps", false, "Don't start linked services.")
-	flags.StringArrayVarP(&opts.volumes, "volumes", "v", []string{}, "Bind mount a volume.")
+	flags.StringArrayVarP(&opts.volumes, "volume", "v", []string{}, "Bind mount a volume.")
 	flags.StringArrayVarP(&opts.publish, "publish", "p", []string{}, "Publish a container's port(s) to the host.")
 	flags.BoolVar(&opts.useAliases, "use-aliases", false, "Use the service's network useAliases in the network(s) the container connects to.")
 	flags.BoolVar(&opts.servicePorts, "service-ports", false, "Run command with the service's ports enabled and mapped to the host.")
 
+	flags.SetNormalizeFunc(normalizeRunFlags)
 	flags.SetInterspersed(false)
 	return cmd
 }
 
+func normalizeRunFlags(f *pflag.FlagSet, name string) pflag.NormalizedName {
+	switch name {
+	case "volumes":
```

Comment:
Should we really introduce an alias given V2 isn’t GA yet?

Reply:
we already have hundred thousands users, so: yes, we should.

Reply:
Right, but there must be a way to handle deprecation for this then.
Or do we just want to keep this around forever?

Reply:
sure, can use `Flags().MarkDeprecated()`

Reply:
I don’t think that would help here as what this does is just add a «deprecated» mention in the help for a flag.
But here ` volumes `  is not a flag anymore and won’t show in the help anyway.

Never mind, not a big deal, and we can do something after GA if needed.